libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -97,7 +106,57 @@ void AP_OpenDroneID::init()
         return;
     }
 
+    load_UAS_ID_from_persistent_memory();
     _chan = mavlink_channel_t(gcs().get_channel_from_port_number(_mav_port));
+    _initialised = true;
+}
+
+void AP_OpenDroneID::load_UAS_ID_from_persistent_memory()
+{
+    id_len = sizeof(id_str);
+    size_t id_type_len = sizeof(id_type);
+    size_t ua_type_len = sizeof(ua_type);
+    if (hal.util->get_persistent_param_by_name("DID_UAS_ID", id_str, id_len) &&
```

Comment:
shouldn't be scanning flash more than once

Reply:
fixed